pkg/grpc/proxy/handler.go

```diff
@@ -161,6 +165,17 @@ func (s *handler) handler(srv any, serverStream grpc.ServerStream) error {
 		// Do not "defer clientCancel()" yet, in case we need to proxy a stream
 		clientCtx, clientCancel := context.WithCancel(outgoingCtx)
 
+		// should we count each retry as a new request from client for metrics?? is so we'll need to move this inside the
```

Comment:
Retries should be considered as part of the same request, and in fact they increase the latency.

PS: Leave this as comment in the PR rather than in the code please :)

Reply:
Oh ok. Then that's probably fine, we should be consistent